platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -345,42 +295,42 @@ public ITable nonLazyLoading() {
                     i = 0;
                     /* Extracting the value for each column from a single row  */
                     for (ColumnData cd : row) {
-                        if (cd.column().isSimple()) {
-                            cell = (Cell) cd;
-                            cellType = cell.column().cellValueType();
-                            value = cellType.getSerializer().deserialize(cell.value());
-                        } else {
-                            throw new RuntimeException("[Hillview can't convert complex data]");
+                        /* Filter the column to only load the one marked true by columnToLoad[] */
+                        if (columnToLoad[i]){
+                            if (cd.column().isSimple()){
+                                cell = (Cell) cd;
+                                cellType = cell.column().cellValueType();
+                                value = cellType.getSerializer().deserialize(cell.value());
+                            }else{
+                                /* Hillview won't process the complex data */
+                                throw new RuntimeException("[Hillview can't convert complex data]");
+                            }
+                            line[i] = value; 
                         }
-                        line[i] = value; 
                         i++;
                     }
-
                     /* Append each line to Hillview column */
                     for (this.currentColumn = 0; this.currentColumn < line.length; this.currentColumn++) {
-                        this.columns[this.currentColumn].append(line[this.currentColumn]);
+                        columns[this.currentColumn].append(line[this.currentColumn]);
                     }
-                    this.currentRow++;
-
-                }else if (unfiltered instanceof RangeTombstoneMarker){
+                } else if (unfiltered instanceof RangeTombstoneMarker) {
                     /* Hillview ignores Tombstone Markers */
                 }
             }
         });
 
-        IColumn[] arrColumns = new IColumn[this.columns.length];
+        List<IColumn> arrColumns = new ArrayList<IColumn>(sizeToLoad);
         IMembershipSet membershipSet = null;
         IAppendableColumn appendableColumn;
         IColumn column;
-        for (int idx = 0; idx < this.columns.length; idx++) {
-            appendableColumn = this.columns[idx];
+        for (int idx = 0; idx < columns.length; idx++) {
```

Comment:
I don't understand what happens here. You should only load the columns you were asked, why are you copying some of them? What's in the other ones?

Reply:
I will remove this. I was using  this.actualSchema.createAppendableColumns(); to create the columns, and it returns IAppendableColumn[] instead of List<IColumn> . The copying is to convert the final result into List. I will update the code to avoid copying